GH Actions add-pr-number fails with SyntaxError when parsing a commit reversed by git revert -m

### Context

Found via: https://github.com/2i2c-org/infrastructure/actions/runs/4344814162

Input: commit message

```
Revert "Merge pull request #2274 from yuvipanda/templebump"

Rollback temporary increase in RAM for temple.

This reverts commit bdcf94c25b9b1e19c58ab9dfe4be35014a806dbd, reversing
changes made to b697f3324b0d0fce0044854ca07003dd9215ecb3.

Fixes #2273
```

https://github.com/2i2c-org/infrastructure/blob/master/.github/workflows/comment-test-link-merged-pr.yaml#L49


Error

```python
  import os
  import re
  
  # Get GITHUB_ENV file in GitHub Actions
  env_file = os.getenv("GITHUB_ENV")
  
  # Retrieve the commit message from GitHub Actions context
  commit_msg = r"""Merge pull request #2[3](https://github.com/2i2c-org/infrastructure/actions/runs/4344814162/jobs/7588658807#step:2:3)07 from 2i2c-org/revert-temple-memory-bump
  
  Revert "Merge pull request #227[4](https://github.com/2i2c-org/infrastructure/actions/runs/4344814162/jobs/7588658807#step:2:4) from yuvipanda/templebump""""
  
  # Use regex to extract the PR number from the commit message
  match = re.search('(?<=#)[0-9]*', commit_msg)
  pr_number = None if match is None else match.group(0)
  
  # Write the continue_workflow and pr_number variables to the
  # GITHUB_ENV file in GitHub Actions
  with open(env_file, "a") as f:
      f.write(f"continue-workflow={'Merge pull request' in commit_msg}")
      f.write("\n")
      f.write(f"pr-number={pr_number}")
  shell: /usr/bin/python {0}
  File "/home/runner/work/_temp/461b9a06-ca30-46e2-8c2e-d912d2dc[5](https://github.com/2i2c-org/infrastructure/actions/runs/4344814162/jobs/7588658807#step:2:5)8a9.py", line 10
    Revert "Merge pull request #22[7](https://github.com/2i2c-org/infrastructure/actions/runs/4344814162/jobs/7588658807#step:2:7)4 from yuvipanda/templebump""""
                                                                 ^
SyntaxError: unterminated string literal (detected at line [10](https://github.com/2i2c-org/infrastructure/actions/runs/4344814162/jobs/7588658807#step:2:10))
Error: Process completed with exit code 1.
```

### Proposal

Should only identify merge requests not reverts

### Updates and actions

_No response_

--- a/.github/workflows/comment-test-link-merged-pr.yaml
+++ b/.github/workflows/comment-test-link-merged-pr.yaml
@@ -42,7 +42,7 @@ jobs:
           env_file = os.getenv("GITHUB_ENV")
 
           # Retrieve the commit message from GitHub Actions context
-          commit_msg = r"""${{ github.event.head_commit.message }}"""
+          commit_msg = r'''${{ github.event.head_commit.message }}'''
 
           # Use regex to extract the PR number from the commit message
           match = re.search('(?<=#)[0-9]*', commit_msg)

